aws/resource_aws_ec2_managed_prefix_list_entry.go

```diff
@@ -0,0 +1,424 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/hashcode"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/ec2/finder"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/ec2/waiter"
+)
+
+func resourceAwsEc2ManagedPrefixListEntry() *schema.Resource {
+	//lintignore:R011
+	return &schema.Resource{
+		Create: resourceAwsEc2ManagedPrefixListEntryCreate,
+		Read:   resourceAwsEc2ManagedPrefixListEntryRead,
+		Update: resourceAwsEc2ManagedPrefixListEntryUpdate,
+		Delete: resourceAwsEc2ManagedPrefixListEntryDelete,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				importParts, err := validateEc2ManagedPrefixListEntryImportString(d.Id())
+				if err != nil {
+					return nil, err
+				}
+				populateEc2ManagedPrefixListEntryFromImport(d, importParts)
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"cidr_block": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IsCIDR,
+			},
+
+			"prefix_list_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(0, 255),
+			},
+
+			"version": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsEc2ManagedPrefixListEntryCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+	pl_id := d.Get("prefix_list_id").(string)
+
+	awsMutexKV.Lock(pl_id)
+	defer awsMutexKV.Unlock(pl_id)
+
+	pl, err := finder.ManagedPrefixListByID(conn, pl_id)
+	if err != nil {
+		return err
+	}
+
+	entry, err := expandPrefixEntry(d, pl)
+	if err != nil {
+		return err
+	}
+	input := &ec2.ModifyManagedPrefixListInput{
+		PrefixListId:   aws.String(pl_id),
+		CurrentVersion: pl.Version,
+	}
+
+	input.AddEntries = []*ec2.AddPrefixListEntry{(*ec2.AddPrefixListEntry)(entry)}
+
+	_, err = conn.ModifyManagedPrefixList(input)
+	if err != nil {
+		return fmt.Errorf("error adding EC2 Managed Prefix List entry (%s): %w", d.Id(), err)
+	}
+	if _, err := waiter.ManagedPrefixListModified(conn, pl_id); err != nil {
+		return fmt.Errorf("error waiting for EC2 Managed Prefix List (%s) update: %w", pl_id, err)
+	}
+
+	getEntriesInput := &ec2.GetManagedPrefixListEntriesInput{
+		PrefixListId: pl.PrefixListId,
+	}
+	var entries []*ec2.PrefixListEntry
+	id := prefixListEntryIDHash(pl_id, entry)
+	log.Printf("[DEBUG] Computed EC2 Managed Prefix List entry ID %s", id)
+
+	err = resource.Retry(5*time.Minute, func() *resource.RetryError {
```

Comment:
Add a `const` for `5*time.Minute` with a descriptive name in `aws/internal/service/ec2/waiter/waiter.go`.